lib/collection/src/lib.rs

```diff
@@ -201,6 +201,20 @@ impl Collection {
             .expect("Shard is guaranteed to be added when id is added to the ring.")
     }
 
+    fn local_shard_by_id(&self, id: ShardId) -> CollectionResult<&LocalShard> {
+        match self.shards.get(&id) {
+            None => Err(CollectionError::service_error(format!(
+                "Shard {} does not exist",
+                id
+            ))),
+            Some(Shard::Remote(_)) => Err(CollectionError::service_error(format!(
+                "Shard {} is not local on peer",
+                id
+            ))),
```

Comment:
`CollectionError::service_error` means instant 500 error, whether the shard moving is expected to be more-or-less standard situation. Might be better to introduce special type of errors

Reply:
Went with a specific error mapped to the `Storage::BadRequest` for a client side error https://github.com/qdrant/qdrant/pull/463/commits/775791872a05a3948288478e90275c2b7608ec30